ambari-common/src/main/python/resource_management/libraries/providers/hdfs_resource.py

```diff
@@ -212,12 +207,17 @@ def __init__(self, hdfs_site, nameservice, run_user, security_enabled, logoutput
     self.run_user = run_user
     self.security_enabled = security_enabled
     self.logoutput = logoutput
-    
+
   @staticmethod
-  def is_webhdfs_available(is_webhdfs_enabled, dfs_type):
-    # only hdfs seems to support webHDFS
-    return (is_webhdfs_enabled and dfs_type.lower() == 'hdfs')
+  def get_default_protocol(default_fs, dfs_type):
+    default_fs_protocol = urlparse(default_fs).scheme.lower()
+    is_viewfs = default_fs_protocol == 'viewfs'
+    return dfs_type.lower() if is_viewfs else default_fs_protocol
     
+  @staticmethod
+  def is_webhdfs_available(is_webhdfs_enabled, default_protocol):
+    return (is_webhdfs_enabled and default_protocol in DFS_WHICH_SUPPORT_WEBHDFS)
```

Comment:
This should have retained the lowercase conversion:

https://github.com/apache/ambari/blob/c36b478ab31ef6c6d9ac1194b1fe5c6a231f50ea/ambari-common/src/main/python/resource_management/libraries/providers/hdfs_resource.py#L219